### PR TITLE
ENH/TST: optimize CDI code,

### PR DESCRIPTION
**Pull request by licode**
The radial_grid from core is for 2D case only, so I still keep the _dist function.

### CODE REVIEW

File: skxray/cdi.py [hunk 1]
```diff
@@ -364,6 +323,9 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
     obj_avg = np.zeros_like(diffracted_pattern).astype(complex)
     avg_i = 0
 
+    obj_a = np.zeros_like(obj_avg)
```
Comment (tacaswell) [inline skxray/cdi.py]: you replace this ref on lines 334, 335, 337

File: skxray/cdi.py [hunk 2]
```diff
@@ -364,6 +323,9 @@ def cdi_recon(diffracted_pattern, sample_obj, sup,
...
+    obj_b = np.zeros_like(obj_avg)
+
     time_start = time.time()
     for n in range(n_iterations):
         obj_old = np.array(sample_obj)
```
Comment (tacaswell) [inline skxray/cdi.py]: do you mutate `sample_obj` in `py_modulus`?

Comment (licode) [inline skxray/cdi.py]: no, that's why I created a new one.

Comment (tacaswell) [inline skxray/cdi.py]: but if you don't mutate it, then you don't need to create a new copy.

Comment (licode) [inline skxray/cdi.py]: Sorry, I mean I mutated sample_obj in py_modulus.

Comment (tacaswell) [inline skxray/cdi.py]: That is a very crucial miss placed negative!

**Comment by licode** [comment]
Other comments on this, @tacaswell ?

**Comment by tacaswell** [comment]
Does it actually go faster?

**Comment by licode** [comment]
Around 5% faster, but the code is much clean.

**Pull request merged**